DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -70,23 +73,29 @@ public IActionResult EditRegistrationPrompt(EditRegistrationPromptViewModel mode
         [Route("Add/SelectPrompt")]
         public IActionResult AddRegistrationPromptSelectPrompt()
         {
-            SetViewBagCustomPromptNameOptions();
-
             var addRegistrationPromptData = TempData.Peek<AddRegistrationPromptData>();
 
             if (addRegistrationPromptData == null || !Request.Cookies.ContainsKey(CookieName))
             {
                 addRegistrationPromptData = new AddRegistrationPromptData();
                 var id = addRegistrationPromptData.Id;
 
-                Response.Cookies.Append(
+                Response.Cookies.Append
+                (
                     CookieName,
                     id.ToString(),
                     new CookieOptions
                     {
                         Expires = DateTimeOffset.UtcNow.AddDays(1)
-                    });
+                    }
+                );
                 TempData.Set(addRegistrationPromptData);
+
+                SetViewBagCustomPromptNameOptions();
+            }
+            else
+            {
+                SetViewBagCustomPromptNameOptions(addRegistrationPromptData.SelectPromptViewModel.CustomPromptId);
             }

```

Comment:
Do we need to separate this from L94? Could we just do:
`SetViewBagCustomPromptNameOptions(addRegistrationPromptData.SelectPromptViewModel?.CustomPromptId);`